plugins/ban.py

```diff
@@ -31,18 +31,27 @@ def ban(self, msg, args):
 
         r = requests.get('https://api.gitter.im/v1/rooms', headers=headers)
         room_data = json.loads(r.text)
+        self.log.info(room_data)
         banned_rooms = []
 
-        for room in filter(lambda x: x.get('uri', None) in joined_rooms,
-                           room_data):
-            if room is not None:
-                url = 'https://api.gitter.im/v1/rooms/' + \
-                    room['id'] + '/bans'
+        if room_data == []:
+            yield 'No rooms found:('
+        else:
+            for room in filter(lambda x: x.get('uri', None) in joined_rooms,
+                               room_data):
+                url = 'https://api.gitter.im/v1/rooms/' + room['id'] + '/bans'
                 rq = requests.post(url, data=data, headers=headers)
+                self.log.info(rq.status_code)
+                self.log.info(rq.json())
```

Comment:
Don't need to log everything. Also, looking at this will not be obvious to where the log came from. The log ideally should have all the information required to pin point itself to the source.